### ISSUE TITLE
BT gamepad support partly broken by recent updates.

### ISSUE THREAD

**Issue by g1t-dlanor** [mupen64plus-ae/mupen64plus-ae#734]
I've had this emulator working perfectly with the same bluetooth gamepads and the same android consoles where it is now failing. And it still works correctly on my tablet, which is possibly related to the fact that this does have a touchscreen, unlike the TV-connected consoles (which use mouse instead of touch interface).

I'm  not sure exactly when this bug appeared but I think it must have been in a fairly recent update. A few months ago it worked perfectly on the consoles too.

The bug should be easy to fix as the program has no problem detecting and correctly using the gamepads in the emulator GUI, both for general GUI usage and (more importantly) when creating or editing the controller profiles. In so doing the emulator shows the correct N64 button response for each mapped BT gamepad button that I press.

But in running any N64 ROM the emulator completely ignores all my gamepad button presses, just as if no controller profile was selected, though it definitely is selected in settings in the same way as on the tablet where this bug does not appear.

**Comment by fzurita** [comment]
Ok, I'm not seeing such issues either on Phones/Tables/Android TV.

In the menus, have you gone to Profiles-->Select Profiles and selected the profile you created for controller 1?

By the way, I haven't touched the controller code in quiet some time, maybe over a year.

**Comment by g1t-dlanor** [comment]
Like I said before, I had definitely selected the controller profile on the failing devices, in exactly the same way as on the non-failing devices. And the problem was repeatable across many reboots and even full app reinstalls followed by full reconfiguration. 

I should also explain that he problem I had does not involve either of the device types you mention, but various TV-connected Android consoles (I have several different brands) that do not use the handicapped 'Android TV' OS variant, which is basically worthless as a gaming platform. Instead they use normal Android OS with normal Android desktop and access to normal 'Google Play' apps (unlike Android TV devices). The main difference between use of a tablet and any of these consoles is the use of  computer mouse instead of touchscreen as the main system GUI interface.

Code changes sometimes have weird side effects on things far beyond what we might expect to be affected at all, and this seems to have been the case here. Note that I said 'have been the case', as some additional changes released lately seem to have fixed the problem.

I've noticed some updates coming after I posted this issue, and today I repeated the same tests as before, with no changes made since the failures. Apparently some of the most recent changes fixed the problem,  because now the controller support of the emulator works fine on all the consoles where it previously failed.

So even though neither one of us knows exactly what happened, GOOD JOB !  :)
And since I no longer have the problem, I'm closing this issue for now.
But I will of course let you know if there is a relapse...
